From 613a88779bf424c9bcae20ccd9c4c04574f37554 Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Fri, 29 Sep 2017 17:08:55 +0200
Subject: [PATCH 4/5] device: fix delay startup complete for unrealized devices

Since commit 6845b9b80a9fcec9d2c9e7b56a37329f38089f2e ("device: delay
startup complete until device is initialized in platform", we also wait
for devices that are still initializing platform/UDEV.

Obviously, that only applies to realized devices.

Otherwise, an unrealized device is going to block startup complete.

Fixes: 6845b9b80a9fcec9d2c9e7b56a37329f38089f2e
(cherry picked from commit 9ad8010fe0c42291580e4a801ed85947ae660d38)
(cherry picked from commit 0ba498b17dc582dcbd9b7102e03496f391d67812)
---
 src/devices/nm-device.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index bacbfb33e..a085e466d 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -12030,7 +12030,8 @@ nm_device_has_pending_action (NMDevice *self)
 	if (priv->pending_actions)
 		return TRUE;
 
-	if (nm_device_get_unmanaged_flags (self, NM_UNMANAGED_PLATFORM_INIT)) {
+	if (   nm_device_is_real (self)
+	    && nm_device_get_unmanaged_flags (self, NM_UNMANAGED_PLATFORM_INIT)) {
 		/* as long as the platform link is not yet initialized, we have a pending
 		 * action. */
 		return TRUE;
-- 
2.13.6

